src/repository/Repository.ts

```diff
@@ -55,16 +55,41 @@ class Repository {
         const guildChannels = guild.channels.valueOf();
         const userList = await guild.members.fetch({user: userIDList});
         const userKarmaList = karmaInfo.map(info => {
-            const userName = userList.get(info.userID).user.username;
+            const user = userList.get(info.userID).user;
             return {
-                user: userName,
+                user,
                 karma: info.karma,
-                channel: guildChannels.get(info.channelID)?.name
+                channel: guildChannels.get(info.channelID)
             }
         });
         return userKarmaList.sort((a, b) => b.karma - a.karma);
     }
 
+    public async getUserKarmaInfo(userID: string): Promise<UserKarmaInfo[] | undefined> {
+        const keys = await this.getKeysAsync(`karmaInfo:${userID}:*`);
+        const karmaInfo: KarmaInfo[] = await Promise.all(keys.map(async (key) => {
+            const karma = await this.getAsync(key);
+            const [_, userID, channelID] = key.split(':');
```

Comment:
This logic assumes that key has a specific syntax separated by colons.

Ideally, we should have some utility functions that abstracts that logic away - that given a key string, it can return `userID` and `channelID`.

Basically, strategy pattern.

Then tomorrow, if we need to change the key to include one more factor, or reorder these, the changes would be minimal - we just need to update the utility.